src/components/Homepage.jsx

```diff
@@ -11,23 +12,53 @@ const Homepage = () => {
       .get("https://nc-news-czlb.onrender.com/api/articles")
       .then((response) => {
         setArticles(response.data);
+        console.log(response.data);
       });
   }, []);
+
+  const sortByDate = () => {
+    let sortedArticles = articles.sort((a, b) => {
+      return b.created_at - a.created_at;
+    });
+    setArticles([...sortedArticles]);
+  };
+
+  const sortByCommentCount = () => {
+    let sortedArticles = articles.sort((a, b) => {
+      return b.comment_count - a.comment_count;
+    });
+    setArticles([...sortedArticles]);
+  };
+
+  const sortByVotes = () => {
+    let sortedArticles = articles.sort((a, b) => {
+      return b.votes - a.votes;
+    });
+    setArticles([...sortedArticles]);
```

Comment:
Hey Harry, rather than using sort in these 3 functions here, you can actually query the API in order to make things a lot easier. I'm not sure if this is the reason why your sort by Date isn't working, but I'd firstly take a look at refactoring the API call to use axios params which should hopefully help 👍